### PR TITLE
[Fix](inverted index) fix memory leak in inverted index writer for array values

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

Issue Number: close #xxx

Add ownership flag to Field's TokenStream value and Analyzer's Reader value, free them when tokenizer finalize.

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp [hunk 1]
```diff
@@ -404,12 +404,15 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
                             _parser_type != InvertedIndexParserType::PARSER_NONE) {
                             // in this case stream need to delete after add_document, because the
                             // stream can not reuse for different field
+                            bool own_token_stream = true;
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:407]: Only used once

**Review by xiaokang — COMMENTED** [review]
pls add a testcase

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp (hunk 1 above)
Comment (airborne12) [reply be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:407]: yes, to make code more readable.

Comment (airborne12) [reply be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:407]: by the way, no need more case, p0 case can cause the memory leak.

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp [hunk 2]
```diff
@@ -421,7 +424,6 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
                     // if this array is null, we just ignore to write inverted index
                     RETURN_IF_ERROR(add_document());
                     _doc->clear();
-                    _CLDELETE(ts);
```
Comment (amorynan) [inline be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:424]: why delete here will make memory leak?

Comment (amorynan) [inline be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:424]: why delete here will make memory leak?

**Review by zzzxl1993 — APPROVED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp (hunk 2 above)
Comment (airborne12) [reply be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:424]: The Field now takes ownership of the TokenStream, eliminating the need for manual deletion.

Comment (airborne12) [reply be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:424]: The field now takes ownership of the TokenStream, eliminating the need for manual deletion.

**Review by qidaye — DISMISSED** [review]

**Comment by qidaye** [comment]
run  buildall

**Comment by airborne12** [comment]
run buildall

**Review by qidaye — APPROVED** [review]

**Pull request merged**